shipment_advice_bill_auto_complete/models/account_move.py

```diff
@@ -100,3 +100,43 @@ def _get_purchase_quantity_from_stock_move_line(self, po_line, move_lines):
         if move_lines.product_id == po_line.product_id:
             quantity_done = sum(move_lines.mapped("qty_done"))
         return quantity_done
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        moves = super().create(vals_list)
+        for move in moves:
+            if move.move_type not in ("in_invoice", "out_refund"):
+                continue
+            shipment = move.line_ids.mapped("shipment_advice_id")
+            if not shipment:
+                continue
+            refs = [
+                "<a href=# data-oe-model=shipment.advice data-oe-id=%s>%s</a>"
+                % tuple(name_get)
+                for name_get in shipment.name_get()
+            ]
+            message = _("This vendor bill has been created from: %s") % ",".join(refs)
+            move.message_post(body=message)
+        return moves
+
+    def write(self, vals):
+        old_shipment = [move.mapped("line_ids.shipment_advice_id") for move in self]
```

Comment:
```suggestion
        old_shipment = {move.id: move.mapped("line_ids.shipment_advice_id") for move in self}
```
with 99% probability, the order will be the same but I don't know if that is guaranteed

Reply:
I would think the same order is guaranteed, the same trick is used in Odoo source code.